standards/core/sync.md

```diff
@@ -0,0 +1,131 @@
+---
+title: WAKU-SYNC
+name: Waku Sync
+editor: Simon-Pierre Vivier <simvivier@status.im>
+contributors:
+ - Prem Chaitanya Prathi <prem@status.im>
+ - Hanno Cornelius <hanno@status.im>
+---
+
+## Abstract
+This specification explains the `WAKU-SYNC` protocol
+which enables the reconciliation of two sets of message hashes
+in the context of keeping multiple Store nodes synchronized.
+Waku Sync is a wrapper around
+[Negentropy](https://github.com/hoytech/negentropy) a [range-based set reconciliation protocol](https://logperiodic.com/rbsr.html).
+
+## Specification
+
+**Protocol identifier**: `/vac/waku/sync/1.0.0`
+
+### Terminology
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The term Negentropy refers to the protocol of the same name.
+Negentropy payload refers to
+the messages created by the Negentropy protocol.
+Client always refers to the initiator
+and the server the receiver of the first payload.
+
+### Design Requirements
+Nodes enabling Waku Sync SHOULD
+manage and keep message hashes in a local cache
+for the range of time
+during which synchronization is required.
+Nodes SHOULD use the same time range,
+for Waku we chose one hour as the global default.
+Waku Relay or Light Push protocol MAY be enabled
+and used in conjunction with Sync
+as a source of new message hashes
+for the cache.
+
+Nodes MAY use the Store protocol
+to request missing messages once reconciliation is complete
+or to provide messages to requesting clients.
+
+### Payload
+
+```protobuf
+syntax = "proto3";
+
+package waku.sync.v1;
+
+message SyncPayload {
+  optional bytes negentropy = 1;
+
+  repeated bytes hashes = 20;
+}
+```
+
+### Session Flow
+A client initiate a session with a server
+by sending a `SyncPayload` with
+only the `negentropy` field set to it.
+This field MUST contain
+the first negentropy payload
+created by the client
+for this session.
+
+The server receives a `SyncPayload`.
+A new negentropy payload is computed from the received one.
+The server sends back a `SyncPayload` to the client.
+
+The client receives a `SyncPayload`.
+A new negentropy payload OR an empty one is computed.
+If a new payload is computed then
+the exchanges between client and server continues until
+the client computes an empty payload.
```

Comment:
What would an empty payload imply? That all hash differences has been exchanged and reconciled?

Reply:
Yes exactly